### PR TITLE
Move CI from Travis to Github Actions

### PR DESCRIPTION
**Pull request by willis102**
Rename the `travis-generate-docs.py` script to `actions-generate-docs.py` and update it to be compatible with Github actions.

### CODE REVIEW

**Comment by willis102** [comment]
We will need to add a `docs-build` environment with the appropriate secrets to this repo, however I don't have admin privileges for it.  It will need to be the same as the environment for `developer.shotgunsoftware.com`.  This should resolve the current failure in CI.

File: .github/workflows/generate-preview.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: Generate Preview
```
Comment (jfboismenu) [inline .github/workflows/generate-preview.yml:8]: Could you add the copyright notice please?

File: .github/workflows/publish-docs.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+name: Publish Docs
```
Comment (jfboismenu) [inline .github/workflows/publish-docs.yml:8]: Could you add the copyright notice please?

**Review by jfboismenu — APPROVED** [review]
Looks good to me. Nice job.

**Comment by ghost** [comment]
[Documentation Preview](http://sg-devdocs.s3-website-us-east-1.amazonaws.com/tk-doc-generator/b00ea677f95a414c67b7938c2d9b67e14352dcf9/index.html)

**Comment by ghost** [comment]
[Documentation Preview](http://sg-devdocs.s3-website-us-east-1.amazonaws.com/tk-doc-generator/8c66cfe5a8be2fe54bec1948a9f6b6d4d3e10ede/index.html)

**Comment by ghost** [comment]
[Documentation Preview](http://sg-devdocs.s3-website-us-east-1.amazonaws.com/tk-doc-generator/8c66cfe5a8be2fe54bec1948a9f6b6d4d3e10ede/index.html)

**Pull request merged**